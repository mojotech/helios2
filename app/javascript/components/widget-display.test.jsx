import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import {
  WidgetDisplay,
  LoadingDisplay,
  DisconnectedDisplay,
} from '@components/widget-display';

const widgets = {
  enabled: [
    {
      id: 2,
      name: 'Weather',
      sidebarText: 'Weather',
    },
    {
      id: 3,
      name: 'Twitter',
      sidebarText: '@MojoTech',
    },
    {
      id: 4,
      name: 'Numbers',
      sidebarText: 'MojoTech by the Numbers',
    },
  ],
  byIdOrFirst: {
    durationSeconds: 20,
    id: 2,
    name: 'Weather',
    showWeather: false,
    sidebarText: 'Weather',
  },
};

const renderer = new ShallowRenderer();

const renderWidgetDisplay = () =>
  renderer.render(
    <WidgetDisplay
      keyToWidgetAction={() => {}}
      clickToWidgetAction={() => {}}
      isPaused={false}
      cityName="Providence"
      widgets={widgets}
    />,
  );

const renderLoadingDisplay = () =>
  renderer.render(<LoadingDisplay cityName="Providence" />);

const renderDisconnectedDisplay = () =>
  renderer.render(<DisconnectedDisplay cityName="Providence" />);

describe('WidgetDisplay component', () => {
  it('renders correctly', () => {
    expect(renderWidgetDisplay()).toMatchSnapshot();
  });
});

describe('LoadingDisplay component', () => {
  it('renders correctly', () => {
    expect(renderLoadingDisplay()).toMatchSnapshot();
  });
});

describe('DisconnectedDisplay component', () => {
  it('renders correctly', () => {
    expect(renderDisconnectedDisplay()).toMatchSnapshot();
  });
});
