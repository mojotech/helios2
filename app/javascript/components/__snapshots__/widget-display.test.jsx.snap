// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DisconnectedDisplay component renders correctly 1`] = `
<styled.div>
  <FullPanel>
    <DisconnectedMessage />
  </FullPanel>
  <Memo
    cityName="Providence"
    error={true}
    isPaused={false}
    selectedWidgetId={0}
    showWeather={false}
    totalTime={0}
    widgets={Array []}
  />
</styled.div>
`;

exports[`LoadingDisplay component renders correctly 1`] = `
<styled.div>
  <FullPanel>
    <LoadingMessage />
  </FullPanel>
  <Memo
    cityName="Providence"
    isPaused={false}
    loading={true}
    selectedWidgetId={0}
    showWeather={false}
    totalTime={0}
    widgets={Array []}
  />
</styled.div>
`;

exports[`WidgetDisplay component renders correctly 1`] = `
<styled.div
  onClick={[Function]}
  onKeyDown={[Function]}
  tabIndex="0"
>
  <FullPanel>
    <getWeatherQuery
      cityName="Providence"
      startTimer={[Function]}
    />
  </FullPanel>
  <Memo
    cityName="Providence"
    error={false}
    isPaused={false}
    loading={true}
    location={
      Object {
        "widgets": Object {
          "byIdOrFirst": Object {
            "durationSeconds": 20,
            "id": 2,
            "name": "Weather",
            "showWeather": false,
            "sidebarText": "Weather",
          },
          "enabled": Array [
            Object {
              "id": 2,
              "name": "Weather",
              "sidebarText": "Weather",
            },
            Object {
              "id": 3,
              "name": "Twitter",
              "sidebarText": "@MojoTech",
            },
            Object {
              "id": 4,
              "name": "Numbers",
              "sidebarText": "MojoTech by the Numbers",
            },
          ],
        },
      }
    }
    selectedWidgetId={2}
    showWeather={false}
    totalTime={20000}
    widgets={
      Array [
        Object {
          "id": 2,
          "name": "Weather",
          "sidebarText": "Weather",
        },
        Object {
          "id": 3,
          "name": "Twitter",
          "sidebarText": "@MojoTech",
        },
        Object {
          "id": 4,
          "name": "Numbers",
          "sidebarText": "MojoTech by the Numbers",
        },
      ]
    }
  />
</styled.div>
`;
